azure/services/scalesetvms/client.go

```diff
@@ -151,10 +146,10 @@ func (ac *azureClient) DeleteAsync(ctx context.Context, resourceGroupName, vmssN
 	}
 
 	return &infrav1.Future{
-		Type:          DeleteFuture,
+		Type:          infrav1.DeleteFuture,
 		ResourceGroup: resourceGroupName,
-		Name:          vmssName,
-		FutureData:    base64.URLEncoding.EncodeToString(jsonData),
+		Name:          instanceID,
+		Data:          base64.URLEncoding.EncodeToString(jsonData),
```

Comment:
should we also have `serviceName` for GetLongRunningOperation to work?

Reply:
changed to using the converter